cylc/rose/utilities.py

```diff
@@ -189,10 +189,18 @@ def rose_config_exists(
 
     # If _any_ of the following are true we want to return True.
     if (
-        Path(srcdir, 'rose-suite.conf').is_file() or
-        opts and opts.opt_conf_keys or
-        opts and opts.defines or
-        opts and opts.rose_template_vars
+        Path(srcdir, 'rose-suite.conf').is_file()
+        or (
+            opts and hasattr(opts, 'opt_conf_keys')
+            and opts.opt_conf_keys
+        )
+        or (
+            opts and hasattr(opts, 'defines')
+            and opts.defines
+        ) or (
+            opts and hasattr(opts, 'rose_template_vars')
+            and opts.rose_template_vars
+        )
```

Comment:
Can you explain this bit. If there's no `rose-suite.conf` file then there is nothing for these options to apply to?

Reply:
It should probably be if there is a rose suite.conf and ...

What do you think should happen if someone uses the rose options (-D and -S, -O is useless) to set template variables without there being a rose-suite.conf present? Error? Why shouldn't you be able to set rows variables on the command line only if you want?

Reply:
I'm not sure you have resolved Oliver's question

Reply:
@oliver-sanders  - From the context I think that removing the check immediately above this block for `rose-suite.conf` (which _is_ happening in Rose anyway so that condition was never used) answers your query - or have I missed something.

@MetRonnie  - Are you suggesting that the check for `rose-suite.conf` should remain and be corrected?

Reply:
Sorry, I'm still confused.

I don't see how the `cylc install` options are relevant here, should this be:

```python
def rose_config_exists(srcdir: Union[Path, str, None], opts: 'Values'):
    return Path(srcdir, 'rose-suite.conf').is_file()
```